### DISCUSSION TITLE
Option to delete site directory after gh-deploy

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#2760] [Ideas]
One thing I currently don't quite understand is why MkDocs keeps a `site` directory after each `gh-deploy` task.
I can understand that it needs to exist in order to push the files into the `gh-pages` branch followed by the push to the remote, but why does it keep the files around after this?

It would be nice if MkDocs could receive an option such as `--delete-site-folder` which when provided would automatically delete the `site` directory after everything has been pushed successfully.

Yes, you can delete the folder yourself, but I feel like the `gh-deploy` shouldn't make us have extra work, if we want to push the sources after the initial deployment.

**Comment by pawamoy** [comment]
I think there's an option to overwrite the git history every time, which is a similar solution to what you're asking (just in reverse order).
Ah, it's not merged yet: https://github.com/mkdocs/mkdocs/pull/2594

**Reply by oprypin** [reply]
@pawamoy Actually could you review #2594?  I don't understand what it does; I never used gh-deploy anyway...
